zebra-state/src/memory_state.rs

```diff
@@ -1,3 +1,420 @@
-pub struct MemoryState {
-    // TODO
+#![allow(dead_code)]
+use std::{
+    cmp::Ordering,
+    collections::BTreeSet,
+    collections::{BTreeMap, HashMap, HashSet},
+    ops::Deref,
+    sync::Arc,
+};
+
+use zebra_chain::{
+    block::{self, Block},
+    primitives::Groth16Proof,
+    sapling, sprout, transaction, transparent,
+    work::difficulty::PartialCumulativeWork,
+};
+
+use crate::service::QueuedBlock;
+
+/// The state of the chains in memory, incuding queued blocks.
+#[derive(Debug, Default)]
+pub struct NonFinalizedState {
```

Comment:
@teor2345 I applied your change to split up the queued blocks here but I went ahead and switch it and the `SledState` to different names. lmk what you think

Reply:
Using the terms from the RFC is great!